src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs

```diff
@@ -0,0 +1,179 @@
+			_contentResolver = Android.App.Application.Context.ContentResolver;
+			_cursor = _contentResolver.Query(oUri,
+									androColumns.ToArray(),  // columns in result
+									null,   // where
+									null,   // where params
+									sortMode);
+
+			if (_cursor is null || _cursor.IsAfterLast)
+			{
+				return entriesList;
+			}
+
+			if (!_cursor.MoveToFirst())
+			{
+				return entriesList;
+			}
+
+			// optimization
+			int _colAllDay, _colLocation, _colStartTime, _colSubject, _colOrganizer, _colDetails, _colCalId, _colId, _colEndTime;
+			_colAllDay = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.AllDay);
+			_colLocation = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.EventLocation);
+			_colStartTime = _cursor.GetColumnIndex(Android.Provider.CalendarContract.Instances.Begin);
+			_colEndTime = _cursor.GetColumnIndex(Android.Provider.CalendarContract.Instances.End);
+			_colSubject = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.Title);
+			_colOrganizer = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.Organizer);
+			_colDetails = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.Description);
+			_colCalId = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.CalendarId);
+			_colId = _cursor.GetColumnIndex("_id");
+
+
+			// odczytanie
+
+			for (uint pageGuard = options.MaxCount; pageGuard > 0; pageGuard--)
+			{
+				var entry = new Appointment();
+
+				// two properties always present in result
+				entry.CalendarId = _cursor.GetString(_colCalId);
+				entry.LocalId = _cursor.GetString(_colId);
+
+				// rest of properties can be switched off (absent in result set)
+				if (_colAllDay > -1)
+				{
+					entry.AllDay = (_cursor.GetInt(_colAllDay) == 1);
+				}
+				if (_colDetails > -1)
+				{
+					entry.Details = _cursor.GetString(_colDetails);
+				}
+				if (_colLocation > -1)
+				{
+					entry.Location = _cursor.GetString(_colLocation);
+				}
+
+				if (_StartTimeRequested)
+				{
+					entry.StartTime = DateTimeOffset.FromUnixTimeMilliseconds(_cursor.GetLong(_colStartTime));
+				}
+
+				if (_DurationRequested)
+				{
+					// calculate duration from start/end, as Android Duration field sometimes is null, and is in hard to parse RFC2445 format 
+					long startTime, endTime;
+					startTime = _cursor.GetLong(_colStartTime);
+					endTime = _cursor.GetLong(_colEndTime);
+					entry.Duration = TimeSpan.FromMilliseconds(endTime - startTime);
+				}
+
+
+				if (_colSubject > -1)
+				{
+					entry.Subject = _cursor.GetString(_colSubject);
+				}
+
+				if (_colOrganizer > -1)
+				{
+					var organ = new AppointmentOrganizer();
+					organ.Address = _cursor.GetString(_colOrganizer);
+					entry.Organizer = organ;
+				}
+
+				entriesList.Add(entry);
+
+				if (!_cursor.MoveToNext())
+				{
+					break;
+				}
+			}
+
+			_cursor.Close();

```

Comment:
It would be best to place this in a `finally`

Reply:
added `using`